wasi/wasi.go

```diff
@@ -1359,11 +1360,19 @@ func newSnapshotPreview1(ctx context.Context) *snapshotPreview1 {
 	return &snapshotPreview1{sys: &defaultSys{}}
 }
 
-func sysCtx(m api.Module) *wasm.SysContext {
```

Comment:
maybe keep sysCtx for the FS unrelated ops

Reply:
I kept it at the beginning, but most of the logic is the same as `fsCtx()` and it looks redundant. `sysFSCtx` returns both and we can just ignore `fsCtx`. Do you prefer having `sysCtx`, `fsCtx` and `sysFSCtx` though they are duplicated?

Reply:
no strong opinion, if anything sysCtx and the double one. We don't benchmark calls to env or args anyway, and if it were a problem we could refactor it later.

Reply:
If we don't export `FSContext`, how can we extract `FSContext` from `CallContext` in `fsCtx`? We cannot access it like

```
func fsCtx(ctx context.Context, m api.Module) *wasm.FSContext {
	if internal, ok := m.(*wasm.CallContext); !ok {
		panic(fmt.Errorf("unsupported wasm.Module implementation: %v", m))
	} else {
		// Override FSContext when it is passed via context
		if fsValue := ctx.Value(expfs.FSKey{}); fsValue != nil {
			fsCtx, ok := fsValue.(*wasm.FSContext)
			if !ok {
				panic(fmt.Errorf("unsupported fs key: %v", fsValue))
			}
			return fsCtx
		}
		return internal.Sys.fs // Can't access `fs` as it is private.
	}
}
```

Do you want to define `type FSContext interface` and return `SysContext` as `FSContext`?

Reply:
I've added `FS()`.
https://github.com/tetratelabs/wazero/pull/571/commits/8e361970044201633d8fbcb6653f11b65462b9c7